### ISSUE TITLE
Misleading exception on https proxy issues

### ISSUE THREAD

**Issue by plockaby** [urllib3/urllib3#2160]
### Subject

Hello, when using a proxy server, if the remote web site is HTTP and it is down urllib3 does the correct thing and returns a connection error. However, if the remote web site is HTTPS then urllib3 reports that the proxy server itself is down when this is incorrect.

### Environment

I'm doing my testing on MacOS but I've experienced this with urllib3 versions 1.24.1 and 1.26.3.

OS Darwin-19.6.0-x86_64-i386-64bit
Python 3.7.3
urllib3 1.26.3

### Steps to Reproduce

First, to illustrate that the proxy server is returning the correct information for urllib3.

This is what happens when requesting a invalid HTTP site:

```
$ curl -sD - -o /dev/null -x http://myproxy.example.com:3128 http://invalid.example.com/
HTTP/1.1 503 Service Unavailable
Server: squid/4.6
Mime-Version: 1.0
Date: Sat, 13 Feb 2021 12:21:05 GMT
Content-Type: text/html;charset=utf-8
Content-Length: 3774
X-Squid-Error: ERR_DNS_FAIL 0
Vary: Accept-Language
Content-Language: en
X-Cache: MISS from myproxy.example.com
X-Cache-Lookup: MISS from myproxy.example.com:3128
Via: 1.1 myproxy.example.com (squid/4.6)
Connection: close
```

This is what happens when requesting an invalid HTTPS site:

```
$ curl -sD - -o /dev/null -x http://myproxy.example.com:3128 https://invalid.example.com/
HTTP/1.1 503 Service Unavailable
Server: squid/4.6
Mime-Version: 1.0
Date: Sat, 13 Feb 2021 12:20:12 GMT
Content-Type: text/html;charset=utf-8
Content-Length: 3762
X-Squid-Error: ERR_DNS_FAIL 0
Vary: Accept-Language
Content-Language: en
```

When using urllib3 to request an invalid HTTP site, I get this expected response:

```
http = urllib3.ProxyManager("http://myproxy.example.com:3128")
r = http.request("GET", "http://invalid.example.com/", retries=False)
print(r.headers)

HTTPHeaderDict({'Server': 'squid/4.6', 'Mime-Version': '1.0', 'Date': 'Sat, 13 Feb 2021 12:23:14 GMT', 'Content-Type': 'text/html;charset=utf-8', 'Content-Length': '3735', 'X-Squid-Error': 'ERR_DNS_FAIL 0', 'Vary': 'Accept-Language', 'Content-Language': 'en', 'X-Cache': 'MISS from myproxy.example.com', 'X-Cache-Lookup': 'MISS from myproxy.example.com:3128', 'Via': '1.1 myproxy.example.com (squid/4.6)', 'Connection': 'close'})

```
However, when using urllib3 to request an invalid HTTPS site, I get this completely unexpected response:

```
http = urllib3.ProxyManager("http://myproxy.example.com:3128")
r = http.request("GET", "https://invalid.example.com/", retries=False)
print(r.headers)

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 594, in urlopen
    self._prepare_proxy(conn)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 807, in _prepare_proxy
    conn.connect()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 308, in connect
    self._tunnel()
  File "/usr/lib/python3.7/http/client.py", line 921, in _tunnel
    message.strip()))
OSError: Tunnel connection failed: 503 Service Unavailable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 13, in <module>
    r = http.request("GET", url, retries=False)
  File "/usr/lib/python3/dist-packages/urllib3/request.py", line 68, in request
    **urlopen_kw)
  File "/usr/lib/python3/dist-packages/urllib3/request.py", line 89, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/usr/lib/python3/dist-packages/urllib3/poolmanager.py", line 446, in urlopen
    return super(ProxyManager, self).urlopen(method, url, redirect=redirect, **kw)
  File "/usr/lib/python3/dist-packages/urllib3/poolmanager.py", line 323, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/urllib3/util/retry.py", line 343, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/lib/python3/dist-packages/six.py", line 692, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 594, in urlopen
    self._prepare_proxy(conn)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 807, in _prepare_proxy
    conn.connect()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 308, in connect
    self._tunnel()
  File "/usr/lib/python3.7/http/client.py", line 921, in _tunnel
    message.strip()))
urllib3.exceptions.ProxyError: ('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Unavailable'))
```

### Expected Behavior

It seems to me that since urllib3 is basically getting all of the same information from the proxy that the result should be the same. I should get HTTP headers back saying that the proxy tried and did not succeed instead of the extremely misleading response that my proxy server is unreachable.

### Actual Behavior

It says that the proxy server is unreachable.

**Comment by sethmlarson** [comment]
Thanks for opening this, you're right that this error message is confusing. It looks like we're not distinguishing between receiving a proxy error on connect versus on request being sent and httplib is raising an `OSError` when the `CONNECT` request fails. Maybe we can make a tweak to the message depending on if the connection to the proxy was successful or not.

**Comment by OneMoreZanuda** [comment]
#1917

**Comment by sethmlarson** [comment]
This has been resolved in the 2.0 branch, errors that occur during the proxy connect step are now always wrapped in `ProxyError`. Going to close this one.
